src/components/tables/FieldSelection/Rows.tsx

```diff
@@ -21,6 +21,78 @@ interface RowsProps {
     columnToSort: string;
 }
 
+const constraintTypeSort_ascending = (
+    a: CompositeProjection,
+    b: CompositeProjection
+): number => {
+    // If a and b have constraint types but they are not equal, compare their severity.
+    // The projection with the greater severity should appear first.
+    if (
+        a.constraint &&
+        b.constraint &&
+        a.constraint.type !== b.constraint.type
+    ) {
+        return a.constraint.type - b.constraint.type;
+    }
+
+    // If a and b do not have defined constraint types or their constraint types are equal,
+    // perform an ascending, alphabetic sort on their fields.
+    return a.field.localeCompare(b.field);
+};
+
+const constraintTypeSort_descending = (
+    a: CompositeProjection,
+    b: CompositeProjection
+): number => {
+    // If a and b have constraint types but they are not equal, compare their severity.
+    // The projection with the greater severity should appear first.
+    if (
+        a.constraint &&
+        b.constraint &&
+        a.constraint.type !== b.constraint.type
+    ) {
+        return b.constraint.type - a.constraint.type;
+    }
+
+    // If a and b do not have defined constraint types or their constraint types are equal,
+    // perform an descending, alphabetic sort on their fields.
+    return b.field.localeCompare(a.field);
+};
+
+const constraintTypeSort = (
+    a: CompositeProjection,
+    b: CompositeProjection,
+    sortDirection: SortDirection
+) => {
+    // See if the values start with alphanumeric
+    const aIsAlphabetical = a.field.localeCompare('a') >= 0;
+    const bIsAlphabetical = b.field.localeCompare('a') >= 0;
+
+    // If a is not alpha and b is then return >0 to put b first
+    if (!aIsAlphabetical && bIsAlphabetical) {
+        return 1;
+    }
+
+    // If a is alpha and b isn't then return <0 to put a first
+    if (aIsAlphabetical && !bIsAlphabetical) {
+        return -1;
+    }
```

Comment:
These duplicated lines should be shared. I think we could make it quick and just have both this and `basicSort_string` calling a 3rd function that handles the shared stuff.

Reply:
The thought crossed my mind but I could not arrive at an appropriate default return value for this new, shared function. The value should not be a number but `null` and `undefined` seem equally unsuitable. What would you deem acceptable here?

Reply:
I think we can go with `null`. That way it is easy to test if the function is returning an actual sort value.